src/compiler/parser.ts

```diff
@@ -3215,6 +3305,12 @@ module ts {
                     return parseVoidExpression();
                 case SyntaxKind.LessThanToken:
                     return parseTypeAssertion();
+                case SyntaxKind.AwaitKeyword:
+                    if (inAwaitContext()) {

```

Comment:
I think we need to do something similar ot how we parse yield expressions.  Namely, we can still parse them, even when we're not in an 'Await' context if it's clear that the user must have been trying to write an await expression.  i.e. if you have ```await foo```, that's definitely an await expression (though ```await(foo)``` would not be).  We can then properly handle this code in all hte IDE layers, and we can give a useful error message that you need to be in an async function.